sdk/identity/azure-identity/src/azure_cli_credential.cpp

```diff
@@ -0,0 +1,654 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/identity/azure_cli_credential.hpp"
+
+#include "private/token_credential_impl.hpp"
+
+#include <azure/core/internal/environment.hpp>
+#include <azure/core/platform.hpp>
+
+#include <cctype>
+#include <cstdio>
+#include <stdexcept>
+#include <thread>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#if defined(AZ_PLATFORM_WINDOWS)
+#if !defined(WIN32_LEAN_AND_MEAN)
+#define WIN32_LEAN_AND_MEAN
+#endif
+#if !defined(NOMINMAX)
+#define NOMINMAX
+#endif
+#include <windows.h>
+#else
+#include <errno.h>
+#include <fcntl.h>
+#include <signal.h>
+#include <spawn.h>
+#include <sys/wait.h>
+#include <unistd.h>
+#endif
+
+using Azure::Identity::AzureCliCredential;
+
+using Azure::DateTime;
+using Azure::Core::Context;
+using Azure::Core::_internal::Environment;
+using Azure::Core::Credentials::AccessToken;
+using Azure::Core::Credentials::AuthenticationException;
+using Azure::Core::Credentials::TokenCredentialOptions;
+using Azure::Core::Credentials::TokenRequestContext;
+using Azure::Identity::AzureCliCredentialOptions;
+using Azure::Identity::_detail::TokenCredentialImpl;
+
+namespace {
+void ThrowIfNotSafeCmdLineInput(std::string const& input, std::string const& description)
+{
+  for (auto const c : input)
+  {
+    switch (c)
+    {
+      case ':':
+      case '/':
+      case '.':
+      case '-':
+      case '_':
+      case ' ':
+        break;
+
+      default:
+        if (!std::isalnum(c))
+        {
+          throw AuthenticationException(
+              "AzureCliCredential: Unsafe command line input found in " + description + ": "
+              + input);
+        }
+    }
+  }
+}
+} // namespace
+
+AzureCliCredential::AzureCliCredential(
+    std::string tenantId,
+    DateTime::duration cliProcessTimeout,
+    Core::Credentials::TokenCredentialOptions const& options)
+    : m_tenantId(std::move(tenantId)), m_cliProcessTimeout(std::move(cliProcessTimeout))
+{
+  static_cast<void>(options);
+  ThrowIfNotSafeCmdLineInput(m_tenantId, "TenantID");
+}
+
+AzureCliCredential::AzureCliCredential(AzureCliCredentialOptions const& options)
+    : AzureCliCredential(options.TenantId, options.CliProcessTimeout, options)
+{
+}
+
+AzureCliCredential::AzureCliCredential(TokenCredentialOptions const& options)
+    : AzureCliCredential(
+        AzureCliCredentialOptions{}.TenantId,
+        AzureCliCredentialOptions{}.CliProcessTimeout,
+        options)
+{
+}
+
+std::string AzureCliCredential::GetAzCommand(
+    std::string const& resource,
+    std::string const& tenantId) const
+{
+  ThrowIfNotSafeCmdLineInput(resource, "Resource");
+  std::string command = "az account get-access-token --output json --resource \"" + resource + "\"";
```

Comment:
az has support for AAD 2.0, so you can use `--scopes` here instead of `resource`.